pkg/querier/timeseries_series_set.go

```diff
@@ -0,0 +1,89 @@
+package querier
+
+import (
+	"sort"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// timeSeriesSeriesSet is a wrapper around a client.TimeSeries slice to implement to SeriesSet interface
+type timeSeriesSeriesSet struct {
+	ts []client.TimeSeries
+	i  int
+}
+
+// Timeseries is a type wrapper that implements the storage.Series interface
+type Timeseries struct {
+	series client.TimeSeries
+}
+
+// TimeSeriesSeriesIterator is a wrapper around a client.TimeSeries to implement the SeriesIterator interface
+type TimeSeriesSeriesIterator struct {
+	ts *Timeseries
+	i  int
+}
+
+type byTimeSeriesLabels []client.TimeSeries
+
+func (b byTimeSeriesLabels) Len() int      { return len(b) }
+func (b byTimeSeriesLabels) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byTimeSeriesLabels) Less(i, j int) bool {
+	return labels.Compare(client.FromLabelAdaptersToLabels(b[i].Labels), client.FromLabelAdaptersToLabels(b[j].Labels)) < 0
+}
+
+func newTimeSeriesSeriesSet(series []client.TimeSeries) *timeSeriesSeriesSet {
+	sort.Sort(byTimeSeriesLabels(series))
+	return &timeSeriesSeriesSet{
+		ts: series,
+		i:  -1,
+	}
+}
+
+// Next implements SeriesSet interface
+func (t *timeSeriesSeriesSet) Next() bool { t.i++; return t.i < len(t.ts) }
+
+// At implements SeriesSet interface
+func (t *timeSeriesSeriesSet) At() storage.Series { return &Timeseries{series: t.ts[t.i]} }
+
+// Err implements SeriesSet interface
+func (t *timeSeriesSeriesSet) Err() error { return nil }
+
+// Labels implements the storage.Series interface
+func (t *Timeseries) Labels() labels.Labels {
+	return client.FromLabelAdaptersToLabels(t.series.Labels)
```

Comment:
I would add a comment saying that this conversion is safe, because t.series.Labels is set in ingester by calling `client.FromLabelsToLabelAdapters(series.Labels())`, so labels are guaranteed to be sorted.

Reply:
done.